src/mpid/ch4/src/ch4r_proc.h

```diff
@@ -482,14 +501,28 @@ static inline int MPIDIU_avt_release_ref(int avtid)
 
 static inline int MPIDIU_avt_init()
 {
-    int i;
+    int i, ret;
 
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDIU_AVT_INIT);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDIU_AVT_INIT);
 
+    MPIDI_CH4_Global.avt_mgr.mmaped_size = 8 * 4 * 1024;
     MPIDI_CH4_Global.avt_mgr.max_n_avts = 1;
     MPIDI_CH4_Global.avt_mgr.next_avtid = 0;
     MPIDI_CH4_Global.avt_mgr.n_avts = 0;
+
+    MPIDI_av_table = (MPIDI_av_table_t **)
+                      mmap(NULL, MPIDI_CH4_Global.avt_mgr.mmaped_size,
+                           PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANON, -1, 0);
+    if (MPIDI_av_table == MAP_FAILED) {
+        abort();
+    }
+    MPIDI_CH4_Global.node_map = (MPID_Node_id_t **)
+                                mmap(NULL, MPIDI_CH4_Global.avt_mgr.mmaped_size,
+                                     PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANON, -1, 0);
+    if (MPIDI_CH4_Global.node_map == MAP_FAILED) {
+        abort();
```

Comment:
Agreed. We should use the standard errcheck macros that eventually call `MPID_Abort`.